alicloud/resource_alicloud_cms_sitemonitor.go

```diff
@@ -0,0 +1,193 @@
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+
+	"github.com/terraform-providers/terraform-provider-alicloud/alicloud/connectivity"
+)
+
+func resourceAlicloudCmsSiteMonitor() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAlicloudCmsSiteMonitorCreate,
+		Read:   resourceAlicloudCmsSiteMonitorRead,
+		Update: resourceAlicloudCmsSiteMonitorUpdate,
+		Delete: resourceAlicloudCmsSiteMonitorDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"address": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"task_name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"task_type": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: validation.StringInSlice([]string{SiteMonitorHTTP, SiteMonitorDNS, SiteMonitorFTP, SiteMonitorPOP3,
+					SiteMonitorPing, SiteMonitorSMTP, SiteMonitorTCP, SiteMonitorUDP}, false),
+			},
+			"alert_ids": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"interval": {
+				Type:         schema.TypeInt,
+				Optional:     true,
+				Default:      1,
+				ValidateFunc: validation.IntInSlice([]int{1, 5, 15}),
+			},
+			"isp_cities": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"options_json": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"task_state": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"create_time": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"update_time": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceAlicloudCmsSiteMonitorCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*connectivity.AliyunClient)
+
+	request := cms.CreateCreateSiteMonitorRequest()
+	request.Address = d.Get("address").(string)
+	request.AlertIds = d.Get("alert_ids").(string)
+	request.TaskName = d.Get("task_name").(string)
+	request.TaskType = d.Get("task_type").(string)
+	request.Interval = strconv.Itoa(d.Get("interval").(int))
+	request.IspCities = d.Get("isp_cities").(string)
+	request.OptionsJson = d.Get("options_json").(string)
+
+	_, err := client.WithCmsClient(func(cmsClient *cms.Client) (interface{}, error) {
+		return cmsClient.CreateSiteMonitor(request)
+	})
+	if err != nil {
+		return fmt.Errorf("Creating site monitor got an error: %#v", err)
+	}
+
+	return resourceAlicloudCmsSiteMonitorRead(d, meta)
+}
+
+func resourceAlicloudCmsSiteMonitorRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*connectivity.AliyunClient)
+
+	request := cms.CreateDescribeSiteMonitorListRequest()
+	request.Keyword = d.Get("task_name").(string)
+	request.Scheme = "https"
+
+	raw, err := client.WithCmsClient(func(cmsClient *cms.Client) (interface{}, error) {
```

Comment:
The Describe method should be re-built to a common service method, so that it can be used in the `Read`, `Delete` and `Test` methods.